kernel/configs/fragments/common/base.conf

```diff
@@ -32,14 +31,11 @@ CONFIG_SHMEM=y
 
 # For security...
 CONFIG_RELOCATABLE=y
-# FIXME - check if we should be setting this
-# https://github.com/kata-containers/packaging/issues/483
-#CONFIG_RANDOMIZE_BASE=y
+CONFIG_RANDOMIZE_BASE=y
```

Comment:
@grahamwhaley Enabling KSLR is recommended from a security point of view, but I dont think you can leverage KSM after this is enabled. 
cc @sameo

Reply:
@amshinde thinking about this, KSM works by looking for hash identical pages - so, I suspect KSM will still work with RANDOMIZE_BASE - but, I suspect you cannot do any native address  based sharing (as the page addresses will be different) - but then I'm not sure we do that between guest kernels loaded by the VM anyway?

I had a dig around to see if there were any documented and notable disadvantages to setting CONFIG_RANDOMIZE_BASE. The only real one I can identify as being real/current is that it will hinder trying to debug crashed kernels, as the address map will change between each boot. I think we debug guest kernels so infrequently that the gains from the positive security aspect probably outweigh the negatives of the debug situation.

We probably want to ask @kata-containers/architecture-committee if they have any thoughts here.

Reply:
I remember noticing that the 'crash' tool had something for this (possibly x86 only and not sure how this works on arm and a guest):
commit ad1a44f5d9b9a41d2c43bb386c765913275f05bf
Author: Dave Anderson <anderson@redhat.com>
Date:   Fri Jan 13 15:38:39 2017 -0500

    Fix for support of /proc/kcore as the live memory source in Linux 4.8
    and later x86_64 kernels configured with CONFIG_RANDOMIZE_BASE, which
    randomizes the unity-mapping PAGE_OFFSET value.  Without the patch,
    the crash session fails during session initialization with the error
    message "crash: seek error: kernel virtual address: <address>
    type: page_offset_base".
    (anderson@redhat.com)